definitions/features/sync_plans.rb

```diff
@@ -33,12 +33,24 @@ def sync_plan_ids_by_status(enabled = true, filter_ids = nil)
     feature(:foreman_database).query(query).map { |r| r['id'].to_i }
   end
 
+  def validate_sync_plan_ids(ids)
+    ids_condition = ids.map { |id| "'#{id}'" }.join(',')
+    query = <<-SQL
+      SELECT id FROM katello_sync_plans WHERE id IN (#{ids_condition})
+    SQL
+    feature(:foreman_database).query(query).map { |r| r['id'].to_i }
+  end
+
   def make_disable(ids)
     update_records(ids, false)
   end
 
   def make_enable
-    update_records(data[:disabled], true)
+    # remove ids of sync plans which no longer exist in DB
+    ids = validate_sync_plan_ids(data[:disabled])
+    @data[:disabled] &= ids
```

Comment:
Is this step necessary? I would have thought the line above returned only valid ids already.

Reply:
@data[:disabled] has all IDs from the local storage, which were disabled before (1)
validate_sync_plan_ids returns all IDs of still existing sync plans from the DB (2)

Only sync plans which where disabled before (1) and still exist in the DB (2) can be enabled using hammer in https://github.com/theforeman/foreman_maintain/pull/681/files#diff-32fb2f52b059a8d16272774a6261b7cf27c1ccc7d5ad4ce057a66a2fcbed3719L78

Reply:
Should this only be passing `@data[:disabled]` to `update_records`? If that is the subset of identified disabled intersection with what was found in the database?

Reply:
You were right. Thank you. `@data[:disabled]` is of course the right variable to pass to update_records()

Reply:
>  validate_sync_plan_ids returns all IDs of still existing sync plans from the DB (2)

But you're passing the *current* `data[:disabled]` to `validate_sync_plan_ids` so the return value should be "all sync plan ids from the old data disabled list that still exist in the DB"?

So, wouldn't this be correct too?
```ruby
data[:disabled] = validate_sync_plan_ids(data[:disabled])
update_records(data[:disabled], true)
```

Reply:
also the `data` vs `@data` confuses me :)

(but overall I think this is doing what it's supposed to be doing and we're just talking cosmetics)

Reply:
data vs @data => https://github.com/theforeman/foreman_maintain/blob/master/definitions/features/sync_plans.rb#L90

true :) thanks again.

Reply:
OMGWTFBBQ ;-)

Thanks!